src/baldr/datetime.cc

```diff
@@ -413,261 +415,29 @@ const std::unordered_map<std::string, size_t> tz_name_to_id = {
-    {"Canada/Central", 148},
-    {"Canada/Eastern", 145},
-    {"Canada/Mountain", 63},
-    {"Canada/Newfoundland", 139},
-    {"Canada/Pacific", 146},
-    {"Canada/Saskatchewan", 130},
-    {"Canada/Yukon", 147},
-    {"Chile/Continental", 134},
-    {"Chile/EasterIsland", 356},
-    {"Cuba", 76},
-    {"Egypt", 5},
-    {"Eire", 303},
-    {"Etc/GMT+0", 263},
-    {"Etc/GMT-0", 263},
-    {"Etc/GMT0", 263},
-    {"Etc/Greenwich", 263},
-    {"Etc/Universal", 291},
-    {"Etc/Zulu", 291},
-    {"Europe/Belfast", 311},
-    {"Europe/Bratislava", 320},
-    {"Europe/Busingen", 337},
-    {"Europe/Guernsey", 311},
-    {"Europe/Isle_of_Man", 311},
-    {"Europe/Jersey", 311},
-    {"Europe/Ljubljana", 296},
-    {"Europe/Mariehamn", 305},
-    {"Europe/Nicosia", 206},
-    {"Europe/Podgorica", 296},
-    {"Europe/San_Marino", 322},
-    {"Europe/Sarajevo", 296},
-    {"Europe/Skopje", 296},
-    {"Europe/Tiraspol", 301},
-    {"Europe/Vaduz", 337},
-    {"Europe/Vatican", 322},
-    {"Europe/Zagreb", 296},
-    {"GB", 311},
-    {"GB-Eire", 311},
-    {"GMT", 263},
-    {"GMT+0", 263},
-    {"GMT-0", 263},
-    {"GMT0", 263},
-    {"Greenwich", 263},
-    {"Hongkong", 187},
-    {"Iceland", 243},
-    {"Indian/Antananarivo", 15},
-    {"Indian/Comoro", 15},
-    {"Indian/Mayotte", 15},
-    {"Iran", 225},
-    {"Israel", 192},
-    {"Jamaica", 88},
-    {"Japan", 227},
-    {"Kwajalein", 369},
-    {"Libya", 18},
-    {"Mexico/BajaNorte", 144},
-    {"Mexico/BajaSur", 100},
-    {"Mexico/General", 104},
-    {"NZ", 352},
-    {"NZ-CHAT", 354},
-    {"Navajo", 61},
-    {"PRC", 219},
-    {"Pacific/Johnston", 366},
-    {"Pacific/Midway", 376},
-    {"Pacific/Ponape", 379},
-    {"Pacific/Saipan", 365},
-    {"Pacific/Samoa", 376},
-    {"Pacific/Truk", 355},
-    {"Pacific/Yap", 355},
-    {"Poland", 335},
-    {"Portugal", 310},
-    {"ROC", 222},
-    {"ROK", 218},
-    {"Singapore", 220},
-    {"Turkey", 306},
-    {"UCT", 290},
-    {"US/Alaska", 22},
-    {"US/Aleutian", 21},
-    {"US/Arizona", 121},
-    {"US/Central", 52},
-    {"US/East-Indiana", 78},
-    {"US/Eastern", 110},
-    {"US/Hawaii", 366},
-    {"US/Indiana-Starke", 79},
-    {"US/Michigan", 62},
-    {"US/Mountain", 61},
-    {"US/Pacific", 94},
-    {"US/Pacific-New", 94},
-    {"US/Samoa", 376},
-    {"UTC", 291},
-    {"Universal", 291},
-    {"W-SU", 317},
-    {"Zulu", 291},        // end deprecated timezones 2018d
-    {"America/Nuuk", 69}, // start renamed timezones 2023c
-    {"Europe/Kyiv", 308},
-    {"Pacific/Kanton", 358}, // end renamed timezones 2023c
+    {"America/Ciudad_Juarez", 117 | (1 << 9)}, // new time zone due to DST
+    {"Asia/Qostanay", 214 | (1 << 9)},         // new time zone due to DST
+    {"America/Nuuk", 69},                      // renamed from America/Godthab
+    {"Europe/Kyiv", 308},                      // renamed from Europe/Kiev
+    {"Pacific/Kanton", 358},                   // renamed from Pacific/Enderbury
```

Comment:
these 3 are new bcs we now also include renamed timezones into this map. a renamed timezone is basically: remove 1 time zone and add 1 new time zone with the same index. no need for bit shifting, so we're not imposing any new limitations doing this.